src/general/application/taxonomies/taxonomies-repository.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+namespace Yoast\WP\SEO\General\Application\Taxonomies;
+
+use WP_Taxonomy;
+use Yoast\WP\SEO\General\Application\Taxonomy_Filters\Taxonomy_Filters_Repository;
+use Yoast\WP\SEO\General\Domain\Taxonomies\Taxonomy;
+
+/**
+ * The repository to get taxonomies.
+ */
+class Taxonomies_Repository {
+
+	/**
+	 * The taxonomy filters repository.
+	 *
+	 * @var Taxonomy_Filters_Repository
+	 */
+	private $taxonomy_filters_repository;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param Taxonomy_Filters_Repository $taxonomy_filters_repository The taxonomy filters repository.
+	 */
+	public function __construct(
+		Taxonomy_Filters_Repository $taxonomy_filters_repository
+	) {
+		$this->taxonomy_filters_repository = $taxonomy_filters_repository;
+	}
+
+	/**
+	 * Returns the object of the filtering taxonomy of a content type.
+	 *
+	 * @param string $content_type The content type the taxonomy filters.
+	 *
+	 * @return array<string,string|array<string, string>> The filtering taxonomy of the content type.
+	 */
+	public function get_content_type_taxonomy( string $content_type ): array {
+		// First we check if there's a filter that overrides the filtering taxonomy for this content type.
+		/**
+		 * Filter: 'wpseo_{$content_type}_filtering_taxonomy' - Allows overriding which taxonomy filters the content type.
+		 *
+		 * @param string $filtering_taxonomy The taxonomy that filters the content type.
+		 */
+		$filtering_taxonomy = \apply_filters( "wpseo_{$content_type}_filtering_taxonomy", '' );
```

Comment:
I would love it if we can refactor this (get the correct taxonomies) part to an class in an `infrastructure` folder, This way this `generic` repository does not need a connection to WP. I just gets a interface of `Taxomony_Collector` or something. That has a method of `get_taxonomies` and that returns a `Taxomomy_List` holding all your `Taxonomy` domain objects. This way you decouple this `Application` class from WP but still rely on your domain objects. If we ever want to swap out how you get your taxonomies to a non wp solution (redis cache, whatever) we can just add a different implemention of that class.